src/tainting/Taint.mli

```diff
@@ -36,7 +36,7 @@ type orig =
 type taint = { orig : orig; tokens : tainted_tokens } [@@deriving show]
 
 type taints_to_sink = {
-  taints_with_precondition : taint list * AST_generic.expr;
+  taints_with_precondition : (taint * unit call_trace) list * AST_generic.expr;
```

Comment:
So this `unit call_trace` is the trace from-current-function-to-sink. Needs a comment / record type / or new type to make it more clear.